### PR TITLE
CM-319: Adjust payment submodule from Insurees for PaymentLayer payments

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/CM-319

Screenshot from 2023-09-20 14-16-53

### CODE REVIEW

File: src/components/PaymentInvoiceFilter.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    debouncedOnChangeFilters([
+      {
+        id: filterName,
+        value: !!value ? value : null,
+        filter: `${filterName}: ${value}`,
+      },
+    ]);
+  };
+
+  const onChangeStringFilter =
```
Comment (dborowiecki) [inline src/components/PaymentInvoiceFilter.js:38]: I think we could rename this function to sth like onChangeStringFilterGenerator, as it's used to setup actual filtering function, not adding filter directly.

**Review by dborowiecki — APPROVED** [review]
Works for me, I left one idea for improvement but nothing blocking. Let me know when draft will be ready for merge or please merge it by yourself if necessary.

**Comment by sniedzielski** [comment]
@dborowiecki You can merge PR, PR is open.

File: src/components/PaymentInvoiceFilter.js (hunk 1 above)
Comment (sniedzielski) [reply src/components/PaymentInvoiceFilter.js:38]: Yes, good idea, but due to some another priorities we can improve this later (this onChangeStringFilter is quite common - it is used in another frontend modules, we can think about generic way of creating such utils for filtering)

**Comment by sniedzielski** [comment]
@dborowiecki If you agree to me merging, I will do it. Thanks for checking PR.

**Pull request merged**